### PR TITLE
Adds Create Account flow; Snapshot Download updates

### PR DESCRIPTION
**Pull request by dgca**
This adds the functionality (i.e. it doesn't look super pertty) for the initial create account flow.

It also updates the snapshot download process, so that it's part of the onboarding flow rather than a modal.

### CODE REVIEW

File: main/api/accounts/handleDeleteAccount.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import { AccountFormat } from "@ironfish/sdk";
+
+import { manager } from "../manager";
+
+export async function handleCreateAccount({ name }: { name: string }) {
```
Comment (dgca) [inline main/api/accounts/handleDeleteAccount.ts:5]: handleDeleteAccount*

File: main/api/ironfish/Ironfish.ts [hunk 2]
```diff
@@ -102,13 +94,26 @@ export class Ironfish {
...
       node.logger,
       node.rpc.getRouter(ALL_API_NAMESPACES),
     );
 
+    this.sdkPromise.resolve(sdk);
+    this.fullNodePromise.resolve(node);
     this.rpcClientPromise.resolve(rpcClient);
   }
+
+  async start() {
```
Comment (dgca) [inline main/api/ironfish/Ironfish.ts:107]: I updated the start command so that `init` creates the Ironfish FullNode, so we can query for accounts on startup.

Now, `start` just starts the node.

File: main/api/manager.ts [hunk 3]
```diff
@@ -36,15 +36,24 @@ export class Manager {
...
+    if (accountsResponse.content.accounts.length === 0) {
+      return "onboarding";
+    }
+
+    const statusResponse = await rpcClient.node.getStatus();
+    const headTimestamp = statusResponse.content.blockchain.headTimestamp;
+    const hoursSinceLastBlock = (Date.now() - headTimestamp) / 1000 / 60 / 60;
+
+    // If the last block was more than a week ago, prompt the user to download a snapshot
+    if (hoursSinceLastBlock > 24 * 7) {
```
Comment (dgca) [inline main/api/manager.ts:52]: What timeframe seems reasonable here?

Comment (dguenther) [reply main/api/manager.ts:52]: Maybe let's do two weeks for now. But eventually, since we don't support partial snapshot downloading, it should probably be expressed in terms of what percentage of the chain we have (since at some point we'll reach a chain length where it's quicker to download a week of blocks over p2p rather than via the snapshot)

And I suppose the absolute most accurate way of estimating it would be (chainLength * time to download snapshot block) vs (chainlength - head) * average time to download block over p2p

Comment (dgca) [reply main/api/manager.ts:52]: For sure, will do 2 weeks for now. Right now the snapshot only takes a few minutes to download, but definitely agree that we should change it to be based on the amount of time it would take to sync in the future.

**Review by dguenther — APPROVED** [review]

File: main/api/snapshot/snapshotManager.ts [hunk 4]
```diff
@@ -112,5 +110,6 @@ export class SnapshotManager {
 
     await downloadedSnapshot.replaceDatabase();
     await fsAsync.rm(downloadedSnapshot.file);
+    await node.openDB();
```
Comment (danield9tqh) [inline main/api/snapshot/snapshotManager.ts:113]: Thinking about the pattern we want here for the node. There's 2 steps that require the node database to be open before we start the node: snapshot and getAccounts. 

In this PR we open the node initially in `Ironfish.init` and then pass it to the snapshot to close it and then make sure it's open when it finishes. I could see it this way. I could also see not having the node open initially and then each individual process snapshot and getAccounts create their own node instance, open it and make sure it's close when its finished.

Personally a process opening its own node and making sure to close down at the end makes a little more sense to me than getting an open node and then making sure it remains open at the end. Wdyt?

Comment (danield9tqh) [reply main/api/snapshot/snapshotManager.ts:113]: If each step opened their own node we could remove the `fullNode` promise. But then we'd also have to create a 2 memory clients. One ephemeral one to get the accounts and then another one on `ironfish.start()` So guess there are pros/cons to each approach

Comment (dguenther) [reply main/api/snapshot/snapshotManager.ts:113]: I think I might want to refactor `snapshotManager` to take an Ironfish (the node app class) instance rather than a separate SDK and node. I think my thoughts here are similar to yours, but with different reasoning. It generally seems like it leads to confusing code if we have operations on the node itself (not through the RPC client) spread throughout the codebase, because you have to handle a bunch of node states, like whether the DB is open and whether the node is stopped/started.

Since the DB can only be held by one node at a time, I think the node instance naturally tends towards being a singleton, vs having per-process nodes in e.g. snapshot and getAccounts.

As an aside, we end up needing the fullNode promise for shutting down the node (and I think it’s generally worthwhile to hold an explicit handle to the node, since it’s running in the background anyway)

TL;DR, I think you're right that each step should make sure the node is in the necessary state, rather than the prior step needing to leave the node in a particular state. But I don't really want to slow this PR down unless it's causing bugs -- I figure I can go refactor it pretty easily if we run into issues at that point.

Comment (danield9tqh) [reply main/api/snapshot/snapshotManager.ts:113]: Yea agreed. Talked about this and I think keeping the fullNode as a singleton on Ironfish makes sense

**Review by danield9tqh — APPROVED** [review]
